src/components/DrugsForProteinCardContainer.tsx

```diff
@@ -1,42 +1,31 @@
-import React, { Fragment, FunctionComponent } from "react";
-import { withRouter, RouteComponentProps } from "react-router";
+import React from "react";
+import { useParams } from "react-router";
 import { v1 } from "uuid";
-import useApi from "./UseApi";
+import useApi from "./hooks/UseApi";
 import DrugsCard, { DrugsData } from "./cards/DrugsCard";
-import PageTemplate from "../PageTemplate";
+import PageTemplate from "../layout/PageTemplate";
 import { Context } from "../types/context";
-import PageContainer from "../PageContainer";
-import ProteinContainer from "./ProteinContainer";
 import { baseUrl } from "../config";
 
-const DrugsForProteinCardContainer: FunctionComponent<RouteComponentProps<
-  any
->> = ({ match }) => {
-  const { id } = match.params;
+const DrugsForProteinCardContainer = () => {
+  const { proteinid } = useParams();
   const { data, isLoading } = useApi<{ results: DrugsData[] }>(
-    `${baseUrl}/protein/${id}/drugs`
+    `${baseUrl}/protein/${proteinid}/drugs`
   );
 
   return (
-    <Fragment>
-      <PageContainer
-        leftColumn={<ProteinContainer id={id} />}
-        rightColumn={
-          <PageTemplate
-            context={Context.DRUG}
-            id={id}
-            length={data?.results && data.results.length}
-            isLoading={isLoading}
-          >
-            {data &&
-              data.results.map((item: DrugsData) => (
-                <DrugsCard data={item} key={v1()} />
-              ))}
-          </PageTemplate>
-        }
-      />
-    </Fragment>
+    <PageTemplate
+      context={Context.DRUG}
+      id={proteinid}
+      length={data?.results && data.results.length}
+      isLoading={isLoading}
+    >
+      {data &&
+        data.results.map((item: DrugsData) => (
+          <DrugsCard data={item} key={v1()} />
```

Comment:
Don't use `v1()` for that, at worst if you can't ensure a unique key from the data (`item.name`, or `${item.sourceId}-${item.name}`?) then go for the index in the array

Reply:
that applies for all the `key={v1()}` in the whole codebase